Pontoon docs: update translate.md

flodolo opened the pull request:
Fixes #208

mathjazz commented on src/tools/pontoon/translate.md:
```diff
@@ -62,89 +54,70 @@ There are two types of quality check failures: errors and warnings.
 
-When you upload translations to Pontoon:
+When uploading translations to Pontoon:
 * Existing strings with matching translations will be ignored.
```
Translations matching the ones in Pontoon will be ignored.

(move that item to the bottom of the list)

mathjazz commented on src/tools/pontoon/translate.md:
```diff
@@ -62,89 +54,70 @@ There are two types of quality check failures: errors and warnings.
-When you upload translations to Pontoon:
+When uploading translations to Pontoon:
 * Existing strings with matching translations will be ignored.
 * New strings and new translations will be imported and attributed to the user uploading the file.
```
Translations different from the ones in Pontoon will be imported and attributed to the user uploading the file.

flodolo reviewed (commented):
Thanks @mathjazz for the detailed review.

Applied most of the suggestions to avoid losing them, but haven't worked on the screenshots yet:
- Need to update sidebar list and mass action
- Need to add one for approval UI
- Need to improve existing one if they have borders

https://github.com/mozilla-l10n/localizer-documentation/pull/213/commits/9b89fb0893e2539ba93b6c19534a2e910eec1a2f

This raises a couple of interesting problems/questions that we need to discuss before going forward with this or other reviews.

The first is how far we want to go with restructuring the context of these pages in this phase. You raise a good point at the beginning about the 4 actions, but I feel like fixing that requires a significant overhaul of the page. Do we do this now or later (2nd pass)?

The second is terminology: we use some terms inconsistently, and we would benefit from having a well-defined glossary, that we could also point users to. At the same time, I don't think we should link to that glossary every time we mention one of these terms, for example.

If we all (@bcolsson  and @mathjazz) agree on the fact that a glossary could be useful, I can start one in a separate PR, and keep this PR on hold as we agree on the terminology to use going forward.

Terms just from this doc: translation, suggestion, approved translation, pretranslation, editing space, translation mode, repository, version control system, sync cycle, translation memory.

flodolo commented on src/tools/pontoon/translate.md:
```diff
@@ -1,58 +1,50 @@
+To suggest a translation, the user can input the translation in the editing space and click `SUGGEST`. The suggestion will then be displayed below the editing space and in the sidebar (in case of multiple suggestions, the most recent suggestion will be displayed).
 
-A notification will appear in the window, indicating that your translation has been saved.
+![Sidebar after suggestion](../../assets/images/pontoon/translate/sidebar_suggest.png "Pontoon’s sidebar before and after a suggestion is submitted")
```
Technically, I guess we could use [this approach](https://stackoverflow.com/questions/19331362/using-an-image-caption-in-markdown-jekyll), but I wonder if it's necessary.

bcolsson replied:
I'd recommend using one screenshot with the same source string with the translation taken from the screenshot above it

flodolo commented on src/tools/pontoon/translate.md:
```diff
@@ -62,89 +54,70 @@ There are two types of quality check failures: errors and warnings.
-![Internal check: error](../../assets/images/pontoon/translate/check_error.png)
+![Internal check: error](../../assets/images/pontoon/translate/check_error.png "Example of error")
 
 Errors are detected using the [compare-locales library](https://pypi.org/project/compare-locales/) and Pontoon’s internal checks.
```
On second thoughts, let me remove it.

bcolsson commented on src/tools/pontoon/translate.md:
```diff
@@ -1,58 +1,50 @@
-Even if a translator has permissions to add translations directly, submitting suggestions to be reviewed helps ensure quality. To manually switch to Suggestion Mode, click the Settings gear to the lower-left side of the editing space and click the icon to the left of `Make Suggestions` so that it becomes a green circle. The `SUGGEST` button should then appear to the lower-right side of the editin ...
-
-![Activate make suggestions](../../assets/images/pontoon/translate/activate_suggest.png)
+Even if the user has permissions to add translations directly, submitting suggestions to be reviewed by another person helps ensure quality. To manually switch to *Suggestion Mode*, click the settings icon in the lower-left side of the editing space and select `Make Suggestions`. The `SUGGEST` button will be displayed in the lower-right side of the editing space, replacing the green `SAVE` button ...
```
```suggestion
It is best practice to submit suggestions for review by another person to ensure translation quality. Some users with elevated permissions may have their editor set to submit translations directly instead of submitting suggestions for review. To manually switch to *Suggestion Mode*, click the settings icon in the lower-left side of the editing space and select `Make Suggestions`. The `SUGGEST` button will be displayed in the lower-right side of the editing space, replacing the green `SAVE` button.
```
Suggestion to make the sentence fit a bit better considering the ability to translate directly is introduced later in this document.

bcolsson commented on src/tools/pontoon/translate.md:
```diff
@@ -62,89 +54,64 @@ There are two types of quality check failures: errors and warnings.
 ## Performing batch actions on strings
 
-Translators can perform mass actions on batches of strings by clicking on the square to the left of any string in the sidebar. When clicked, the square will show a checkmark and the editing space will show the mass action window.
+Mass actions can be performed on multiple strings at once at once by clicking on the square to the left of any string in the sidebar. When clicked, the square will show a checkmark and the editing space will show the mass action panel. Note that mass actions are only available to users with [translator rights](users.md#user-roles).
```
```suggestion
Mass actions can be performed on multiple strings at once by clicking on the square to the left of any string in the sidebar. When clicked, the square will show a checkmark and the editing space will show the mass action panel. Note that mass actions are only available to users with [translator rights](users.md#user-roles).
```

bcolsson commented on src/tools/pontoon/translate.md:
```diff
@@ -62,89 +54,70 @@ There are two types of quality check failures: errors and warnings.
-Translators can perform mass actions on batches of strings by clicking on the square to the left of any string in the sidebar. When clicked, the square will show a checkmark and the editing space will show the mass action window.
+It’s possible to perform mass actions on batches of strings by clicking on the square to the left of any string in the sidebar. When clicked, the square will show a checkmark and the editing space will show the mass action window.
 
 ![Mass action window](../../assets/images/pontoon/translate/batch_actions.png)
```
As I'm going through this pass it does indeed seem like its out of date. 
Left sidebar styling differs from how it appears on Pontoon today, and the String Selected / Select All locations are now switched,

flodolo commented:
[last commit](https://github.com/mozilla-l10n/localizer-documentation/pull/213/commits/9bb6775aa08dac7e425c1d519031a31a0b91d565):
* Reorganized the initial part with the structure suggested by @mathjazz
* Redone the screenshots, removed obsolete files that I missed
* Added a first pass for a glossary file, with the terms that we discussed so far

flodolo commented on src/tools/pontoon/translate.md:
```diff
@@ -87,11 +82,11 @@ In the `REPLACE IN TRANSLATIONS` section, the user can input the text to search
 ![Profile dropdown menu](../../assets/images/pontoon/translate/profile_menu.png)
 
-Anyone can download terminology (`.tbx`), translation memory (`.tmx`) and translations, while only translators can upload translations. When downloading translations:
+Anyone can download terminology (`.tbx`), translation memory (`.tmx`) and translations, while only user with translator permissions can upload translations. When downloading translations:
```
Fixed this (user -> users)

flodolo commented on src/tools/pontoon/translate.md:
```diff
@@ -87,11 +82,11 @@ In the `REPLACE IN TRANSLATIONS` section, the user can input the text to search
 ## Downloading and uploading translations
 
-Pontoon provides the ability to download and upload translations, including terminology and translation memories. To access these features,  click on the profile icon in the top-right corner of any page. Note that the user must be in the translation workspace for the download/upload options to be displayed in the dropdown menu.
+Pontoon provides the ability to download and upload translations, including [terminology](glossary.md#terminology) and [translation memories](glossary.md#translation-memory. To access these features, click on the profile icon in the top-right corner of any page. Note that the user must be in the translation workspace for the download/upload options to be displayed in the dropdown menu.
```
Fixed broken link to TM

bcolsson reviewed (approved):
Looks good. Some minor grammar nits and alignment of screenshot text with documentation copy, but don't need to review after those fixes so approving.

bcolsson commented on src/tools/pontoon/glossary.md:
```diff
@@ -0,0 +1,60 @@
+
+## Translation memory
+
+Translation memory is a database that stores all previous translations for a specific string. It can be leverage to provide suggestions when translating new content.
```
```suggestion
Translation memory is a database that stores all previous translations for a specific string. It can be leveraged to provide suggestions when translating new content.
```

flodolo commented on src/tools/pontoon/translate.md:
```diff
@@ -1,58 +1,45 @@
+![Approve button](../../assets/images/pontoon/translate/approve_button.png "Editing space in Pontoon with the green APPROVE button")
 
-To review a suggestion, search for unreviewed strings in the sidebar.
+If the current translation is **not** acceptable, it can be rejected by clicking the reject icon to the right of the suggestion. After rejecting a suggestion or pretranslation, it’s also possible to delete it completely by clicking the trashcan icon. In case of pretranslations, the rejected text will be also removed from version control systems (where applicable).
```
Never knew there was a [difference](https://blog.harwardcommunications.com/2013/12/04/how-to-use-in-case/) 😱

flodolo commented on src/tools/pontoon/translate.md:
```diff
@@ -62,89 +49,64 @@ There are two types of quality check failures: errors and warnings.
+Note that `APPROVE ALL` accepts the latest suggestion, but doesn’t reject other suggestions if available.
 
-Note that mass actions are only available to users with [translators rights](users.md#user-roles).
+In the `REPLACE IN TRANSLATIONS` section, the user can input the text to search for, and the text to replace it with. This is a basic find and replace feature that will work only on the selected strings.
```
Didn't even realize that text changed 🤦🏼

mathjazz commented on src/tools/pontoon/translate.md:
```diff
@@ -62,89 +49,64 @@ There are two types of quality check failures: errors and warnings.
+Note that `APPROVE ALL` accepts the latest suggestion, but doesn’t reject other suggestions if available.
 
-Note that mass actions are only available to users with [translators rights](users.md#user-roles).
+In the `FIND & REPLACE IN TRANSLATIONS` section, the user can input the text to search for, and the text to replace it with. This is a basic find and replace feature that will work only on the selected strings.
```
Nit: Do we need _the_ in "input the text" and "and the text"?

flodolo replied:
I think so. The alternative would be to use `and its replacement`, but it's less clear than the current text.

flodolo commented on src/tools/pontoon/translate.md:
```diff
@@ -1,58 +1,45 @@
-![Sidebar after suggestion](../../assets/images/pontoon/translate/sidebar_suggest.png)
+When the user is in *Translation Mode*, a green `SAVE` button will be displayed instead of `SUBMIT`. When saving a translation:
+* The translation will be displayed directly below the editing space and in the sidebar.
+* The translation will be stored in the version control system (where applicable).
```
DB-projects don't use VCS, that's not applicable for them.

flodolo replied:
Note that this form is used in a few places (on purpose), because explaining it each time would be too much, and omitting it would be imprecise.

mathjazz reviewed (approved):
I have some stylistic suggestions for the Glossary and I've also sent it through a grammar check. 

I found it easier to edit the file directly:
https://gist.github.com/mathjazz/bb44ab663655705e8bfee724e2707d0d

Feel free to make any changes. Other than that this is good to go.

flodolo commented:
Overall it looks good to me.

One question (in case as a follow-up): you removed rejected suggestions from "suggestions". Where do those fall into?
